test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/lang/SystemJSTest.scala

```diff
@@ -57,4 +59,58 @@ class SystemJSTest {
       assertEquals(x1FirstHash, System.identityHashCode(x1))
     }
   }
+
+  @Test def systemProperties(): Unit = {
+    def get(key: String): String = java.lang.System.getProperty(key)
+
+    // Defined in System.scala
+    assertEquals("1.8", get("java.version"))
+    assertEquals("1.8", get("java.vm.specification.version"))
+    assertEquals("Oracle Corporation", get("java.vm.specification.vendor"))
+    assertEquals("Scala.js", get("java.vm.name"))
+    assertEquals("1.8", get("java.specification.version"))
+    assertEquals("/", get("file.separator"))
+    assertEquals(":", get("path.separator"))
+    assertEquals("\n", get("line.separator"))
+    assertNotNull(get("java.vm.version"))
+
+    // Added via __ScalaJSEnv in ScalaJSPluginInternal
+
+    val compliantAsInstanceOf = get("scalajs.compliant-asinstanceofs") == "true"
+    if (compliantAsInstanceOf) {
+      try {
+        Int.box(5).asInstanceOf[String]
+        fail("Invalid class cast succeeded in presence of " +
+            "\"compliant-asinstanceofs\" flag.")
+      } catch {
+        case _: ClassCastException => // As expected
+      }
+    }
+
+    val productionMode = get("scalajs.production-mode") == "true"
+    val developmentMode = get("scalajs.development-mode") == "true"
+    assertTrue(productionMode ^ developmentMode)
+
+    val inNode = get("scalajs.nodejs") == "true"
+    val inPhantomJS = get("scalajs.phantomjs") == "true"
+    val inRhino = get("scalajs.rhino") == "true"
+
+    if (inNode) {
+      val process = js.Dynamic.global.process
+      assertFalse(js.isUndefined(process))
+      assertFalse(inPhantomJS || inRhino)
+    } else if (inPhantomJS) {
+      assertFalse(js.isUndefined(js.Dynamic.global.callPhantom))
+      assertFalse(inNode || inRhino)
+    } else if (inRhino) {
+      // TODO find some platform specific test for Rhino
+      assertFalse(inNode || inPhantomJS)
```

Comment:
Apparently my comment disappeared, so here it is again: you can test for the presence of `js.Dynamic.global.Packages`.